app/validation/validators.py

```diff
@@ -218,6 +219,16 @@ def __call__(self, form, field):
             raise validators.StopValidation(self.message)
 
 
+class YearCheck:
+    def __init__(self, message=None):
+        self.message = message or error_messages['INVALID_DATE']
+
+    def __call__(self, form, field):
+        # Must be 4 digits long
+        if not re.match(r'\d{4}$', str(form.year.data)):
```

Comment:
This would allow `0001` for the year, which isn't ideal?

Reply:
0001 is still a valid 4 digit year - unless the question specifies a minimum date, I can't see why this would be blocked as a date.

Reply:
I've spoken with @charlotte-hirst about this. This is fine as the check is primarily so people don't accidentally enter a 2 digit year without realising. Another card will be raised to consider making this a format validation error rather than an invalid date error.